spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitProperties.java

```diff
@@ -467,6 +483,11 @@ public void setSize(Integer size) {
 	public static class Listener {
 
 		/**
+		 * Container type.
+		 */
+		private ContainerType type = ContainerType.SIMPLE;
```

Comment:
Unfortunately, it's not black and white - each container has certain pros and cons. I feel that changing the container type without the user explicitly choosing  to do so could cause unexpected problems for users. 

I prefer that the user explicitly opt-in to using the new container. [Documentation here](http://docs.spring.io/spring-amqp/docs/2.0.0.BUILD-SNAPSHOT/reference/html/_reference.html#choose-container).